kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriberServiceImpl.java

```diff
@@ -33,7 +43,50 @@ public void add(Subscriber subscriber) {
 
     @Override
     public void update(Subscriber subscriber) {
+
+        // cache the previous version of the subscriber in order to compare before/after
+        Subscriber retrievedSubscriber = subscriberDataService.findByCallingNumber(subscriber.getCallingNumber());
+
         subscriberDataService.update(subscriber);
+
+        // update start dates for subscriptions if reference date (LMP/DOB) has changed
+        Set<Subscription> subscriptions = retrievedSubscriber.getActiveSubscriptions();
+        Iterator<Subscription> subscriptionIterator = subscriptions.iterator();
+        Subscription subscription;
+
+        while (subscriptionIterator.hasNext()) {
+            subscription = subscriptionIterator.next();
+
+            if ((subscription.getSubscriptionPack().getType() == SubscriptionPackType.PREGNANCY) &&
+                    (retrievedSubscriber.getLastMenstrualPeriod() != subscriber.getLastMenstrualPeriod()))  {
+
+                updateOrSubscribe(subscriber, subscription, subscriber.getLastMenstrualPeriod(),
+                        subscriptionService.getSubscriptionPack("pregnancyPack"));
+
+            } else if ((subscription.getSubscriptionPack().getType() == SubscriptionPackType.CHILD) &&
+                    (retrievedSubscriber.getDateOfBirth() != subscriber.getDateOfBirth()))  {
+
+                    updateOrSubscribe(subscriber, subscription, subscriber.getDateOfBirth(),
+                            subscriptionService.getSubscriptionPack("childPack"));
+
+            }
+        }
+    }
+
+    private void updateOrSubscribe(Subscriber subscriber, Subscription subscription, LocalDate referenceDate,
+                                   SubscriptionPack pack) {
+        if (referenceDate == null) {
+            return; // TODO: log?
```

Comment:
I'm not sure how we'd actually get into this situation -- I think it's a bug if we do. So I think either log or throw. Thoughts?